catalog/core/catalog-core-impl/pubsub/src/main/java/ddf/catalog/pubsub/EventProcessorImpl.java

```diff
@@ -436,7 +437,22 @@ public DeleteResponse process(DeleteResponse deleteResponse) throws PluginExecut
     }
 
     public static enum DateType {
-        modified, effective, expiration, created
+        modified, effective, expiration, created, metacard_created(Core.METACARD_CREATED), metacard_modified(
+                Core.METACARD_MODIFIED);
+
+        private String attributeName;
+
+        DateType() {
+            attributeName = this.name();
+        }
+
+        DateType(String attributeName) {
+            this.attributeName = attributeName;
+        }
+
+        public String getAttributeName() {
```

Comment:
❗️ We should add a JavaDoc comment here to explain what this new `attributeName` is. We should also mention in the `enum` JavaDoc comments that `getAttributeName()` should be used instead of `name()` or `toString()` to map to and from metacard attribute names, and that `valueOf()` should will not work with an attribute name.